### DISCUSSION TITLE
osu!mania isn't that comfortable to play in landscape mode on mobile

### DISCUSSION THREAD

**Discussion by 33KK** [ppy/osu#16783] [Ideas]
Recently #16733 was merged, locking the screen orientation to landscape. Mania controls require tapping on the lanes, so those need to be pretty wide. Wider lanes are much shorter, so there's much less screen space for the notes and half of the screen is unused. 20:9 aspect ratio doesn't help too. I think built-in android auto rotation toggle should be more than enough (osu ignores it though?) instead of the hardcoded landscape orientation.

**Comment by bdach** [comment]
known issue. will be addressed by virtual controls or some sort of playfield scaling.

game orientation on mobile has been discussed back and forth too many times already. the general UI just does not work well enough in portrait.

**Comment by KrispyKebaeb** [comment]
Well the vertical should return, but they should make landscape the default orientation which can be switched with android auto rotate, playing osu vertically could be a community challenge, so there might not be a reason to keep it but there isnt a reason to remove it either, they should rework the orientation a bit.

**Reply by Susko3** [reply]
Unfortunately, Android doesn't provide a simple way to default to landscape and honor auto-rotate at the same time. If auto-rotate is disabled Android will default to portrait orientation on phones, something that's not wanted.

~~I suggested having orientation in settings, but peppy seems to be hesitant to adding more settings.~~

**Comment by Susko3** [comment]
If you want to play portrait, at the risk of having less then stellar UX and using a (currently) unsupported aspect ratio, give [Set Orientation](https://www.apkmirror.com/apk/eyes-free-project/set-orientation/set-orientation-1-1-4-release/set-orientation-1-1-4-android-apk-download/) a shot. It can force any app on your phone into any orientation that you want.

I was under the impression that the playfield is always locked to 4:3, so the phone being in portrait would reduce the available area. Is this not the case for osu!mania?

**Reply by frenzibyte** [reply]
I can confirm osu!mania feels slightly narrower on portrait compared to landscape:

<details>
<summary>landscape</summary>

![image](https://user-images.githubusercontent.com/22781491/152702981-fdbffb25-4a58-4a4b-8017-3e989fdc8cb5.png)

</details>

<details>
<summary>portrait</summary>

![image](https://user-images.githubusercontent.com/22781491/152702975-e56ba144-3ab8-4ded-a873-1ca2c59f809c.png)

</details>

(unless what's wanted here is the visible length of the column?)

I believe that virtual controllers are the correct path forward for playing such rulesets (tracked under #4195).

**Comment by smileyhead** [comment]
If UI layout is the problem, I already have a few ideas on how the current UI could be adapted to portrait. A couple of minor changes for when the screen is portrait should be enough.
Also, just an opinion, but I think Android's way of handling auto-rotate is fine. What if I want to play in portrait, but the game wants to prefer landscape? The game would rotate itself automatically, and I would have to rotate my device into landscape and back into portrait for it to reflect my preference.

**Reply by frenzibyte** [reply]
Not really sure that sounds like a solution but more of a workaround, and I wonder if there is any existing app that does such thing. Either the game supports rotating in all four orientations, or only landscape.

**Comment by VladutzNVT** [comment]
just putting the screen in portrait obviously wont be enough, i suggest doing what funky friday did for mobile: the play field is your entire screen but split in 4 big tiles
![WIN_20220418_10_49_34_Pro](https://user-images.githubusercontent.com/103924738/163776161-04a73a7d-b6f6-405f-8895-0a22301dec10.jpg)
so maybe itll look something like this?
![WIN_20220418_10_44_02_Pro](https://user-images.githubusercontent.com/103924738/163776421-6938d5e2-7491-4666-8cf0-5c3ae27c5f85.jpg)
also you dont have to make the entire game portrait, just make mania gameplay portrait (why would i need main menu and stuff to be in portrait???)

**Reply by frenzibyte** [reply]
The worst part about it is that now you have to keep rotating your phone when switching between non-gameplay and gameplay context, which can get worse if you use a stand for your phone.

**Reply by VladutzNVT** [reply]
i think the best mobile options are:
A: landscape with the tile controls i mentioned (or maybe the tiles could be placed on the left and right of your screen so that the highway isnt covered by the controls (but that probably wouldnt work with skins that place the highway on the left/right)
B: what i proposed

both of these should be added

**Reply by RUvlad1** [reply]
the buttons being split to fit the entire screen would be very helpful, even in landscape
something like this maybe
![concept](https://github.com/ppy/osu/assets/60317461/8b08345d-ed24-4050-9f14-097e8e69c9fe)

would be really nice to see this being added to lazer

**Comment by PhilipPlayz** [comment]
how about try how malody approached portrait mode

its actually still on landscape but the lanes are in portrait, its kinda hard to explain but i have a screen record that hopefully you would understand what im trying to explain

https://youtu.be/4ZfJyijVo9E

the only problem i see with this is anything higher than 4k won't be possible

**Reply by frenzibyte** [reply]
Pretty much https://github.com/ppy/osu/discussions/16783#discussioncomment-2584666.

**Comment by AAMBIENTT** [comment]
Just installed lazer on my ipad pro, its literally impossible to play mania on mobile because its so small. would love to see a width slider for the mania board
